projects/baby_driver/inc/i2c_read.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+// This module allows for data to be read from an IC over I2C at a given I2C port and address,
+// can also perform a register read instead of a regular read if specified.
+// Requires dispatcher, interrupts, soft timers, gpio, the event queue, and CAN to be initialized.
+
+#include "i2c.h"
+#include "status.h"
+
+typedef struct I2CReadCommand {
+  I2CPort port;
+  I2CAddress address;
+  uint8_t rx_len;
+  uint8_t is_reg;
+  uint8_t reg;
+} I2CReadCommand;
+
+#define I2C_READ_DEFAULT_TIMEOUT_MS 5
```

Comment:
Similarly, this should be renamed something like `I2C_READ_DEFAULT_TX_DELAY_MS`.